simple/cq_data.c

```diff
@@ -481,7 +481,7 @@ int main(int argc, char **argv)
 
 	hints->ep_attr->type = FI_EP_MSG;
 	hints->caps = FI_MSG;
-	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR;
+	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR | FI_RX_CQ_DATA;
```

Comment:
I think we might be confusing `FI_REMOTE_CQ_DATA` and `FI_RX_CQ_DATA`.  If I understand it correctly, the former is an operation flag to indicate that remote CQ data is to be sent for a transfer, and the latter is a mode bit used to indicate that an application must post a receive operation to catch incoming RDMA_WRITE or atomic immediate data (as is required by the verbs provider).

Also, I didn't catch that this test is using `fi_senddata` and not `fi_writedata`, so the verbs provider issue doesn't apply here since a recv is needed regardless.  So `FI_RX_CQ_DATA` is unneeded in this test.